### ISSUE TITLE
Apply checkstyle rule only to specific module

### ISSUE THREAD

**Issue by pigliamosche** [checkstyle/checkstyle#9002]
**Goal:** apply a checkstyle rule only in a specific module

___

I tried apply a the following checkstyle rule only with a specific module

```java
...
<module name="RegexpSingleline">
        <property name="format" value='string\s+name\s*=\s*"(?!my_string_prefix)\w+"'/>
        <property name="id" value="MyStringPrefix"/>
        <property name="message" value="Use my_string_prefix as a prefix for strings"/>
        <property name="fileExtensions" value="xml"/>
    </module>
...
```
I set the id and in the suppression.xml file the name of the module that I want to exclude with the following regex

```java
...
<suppressions>
  <suppress id="MyStringPrefix"
    files="[\\/]((?!mymodulename).)*[\\/]"/>
</suppressions>
```

it doesn't work as expected
it works only if I explicitly add all the others module names and excluding `mymodulename`, but this is something that I don't like because is not future proof if someone else add a new module in the codebase

could someone please help me?

**Comment by romani** [comment]
@pigliamosche , can you prove that your regexp on module name is valid and work as expected.
Better to prove in CLI by some shell tool, or  ... https://regex101.com/

**Comment by rnveach** [comment]
> `"[\\/]((?!mymodulename).)*[\\/]"`

I think this expression has multiple levels of issues.

1) `[\\\/]` could be matched multiple times if the folder is nested enough, so the not match could be picking up other parts of the path.
Example: `/home/rveach/workspace/folder/mymodulename` could match up on `/home/` as its the first occurrence found.

2) `*` means it could match on nothing which doesn't make sense.
